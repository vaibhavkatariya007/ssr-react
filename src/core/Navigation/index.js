import React from 'react';
import { Button } from 'antd';

const Navigation = ({ prevPage, nextPage, ...rest }) => (
  <div className="custom-pagination">
    <Button type="primary" onClick={prevPage} disabled={!rest.page}>
      Previous
    </Button>
    |
    <Button type="primary" onClick={nextPage} disabled={!rest.nbPages}>
      Next
    </Button>
  </div>
);

export default Navigation;
