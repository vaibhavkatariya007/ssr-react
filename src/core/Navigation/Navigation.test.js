import React from 'react';
import { shallow, mount } from 'enzyme';
import Navigation from './';
describe('Testing Navigation Component', () => {
  const component = shallow(<Navigation />);

  test('Navigation Component exists', () => {
    expect(component.exists()).toBe(true);
  });

  test('Navigation should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  test('Navigation has two buttons', () => {
    const mockCallBack = jest.fn();
    const component = mount(<Navigation />);
    const container = component.find('Button');
    expect(container.length).toEqual(2);
    // container.forEach((button) => {
    //   button.simulate('click');
    //   expect(mockCallBack.mock.calls.length).toEqual(1);
    // });
  });
});
