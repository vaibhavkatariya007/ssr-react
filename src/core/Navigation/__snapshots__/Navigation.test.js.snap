// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navigation Component Navigation should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(InternalButton)
          block={false}
          disabled={true}
          ghost={false}
          htmlType="button"
          loading={false}
          type="primary"
        >
          Previous
        </ForwardRef(InternalButton)>,
        "|",
        <ForwardRef(InternalButton)
          block={false}
          disabled={true}
          ghost={false}
          htmlType="button"
          loading={false}
          type="primary"
        >
          Next
        </ForwardRef(InternalButton)>,
      ],
      "className": "custom-pagination",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": false,
          "children": "Previous",
          "disabled": true,
          "ghost": false,
          "htmlType": "button",
          "loading": false,
          "onClick": undefined,
          "type": "primary",
        },
        "ref": null,
        "rendered": "Previous",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "__ANT_BUTTON": true,
          "defaultProps": Object {
            "block": false,
            "ghost": false,
            "htmlType": "button",
            "loading": false,
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
      "|",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "block": false,
          "children": "Next",
          "disabled": true,
          "ghost": false,
          "htmlType": "button",
          "loading": false,
          "onClick": undefined,
          "type": "primary",
        },
        "ref": null,
        "rendered": "Next",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Group": [Function],
          "__ANT_BUTTON": true,
          "defaultProps": Object {
            "block": false,
            "ghost": false,
            "htmlType": "button",
            "loading": false,
          },
          "displayName": "Button",
          "render": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(InternalButton)
            block={false}
            disabled={true}
            ghost={false}
            htmlType="button"
            loading={false}
            type="primary"
          >
            Previous
          </ForwardRef(InternalButton)>,
          "|",
          <ForwardRef(InternalButton)
            block={false}
            disabled={true}
            ghost={false}
            htmlType="button"
            loading={false}
            type="primary"
          >
            Next
          </ForwardRef(InternalButton)>,
        ],
        "className": "custom-pagination",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": false,
            "children": "Previous",
            "disabled": true,
            "ghost": false,
            "htmlType": "button",
            "loading": false,
            "onClick": undefined,
            "type": "primary",
          },
          "ref": null,
          "rendered": "Previous",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Group": [Function],
            "__ANT_BUTTON": true,
            "defaultProps": Object {
              "block": false,
              "ghost": false,
              "htmlType": "button",
              "loading": false,
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
        "|",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "block": false,
            "children": "Next",
            "disabled": true,
            "ghost": false,
            "htmlType": "button",
            "loading": false,
            "onClick": undefined,
            "type": "primary",
          },
          "ref": null,
          "rendered": "Next",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Group": [Function],
            "__ANT_BUTTON": true,
            "defaultProps": Object {
              "block": false,
              "ghost": false,
              "htmlType": "button",
              "loading": false,
            },
            "displayName": "Button",
            "render": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
